app/assets/javascripts/dashboard/views/dataset/dataset_import_view.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+
+import React from "react";
+import { Button, Spin, Input, Checkbox } from "antd";
+import Request from "libs/request";
+import update from "immutability-helper";
+import Toast from "libs/toast";
+import type { APIDatasetType } from "admin/api_flow_types";
+
+class DatasetImportView extends React.PureComponent {
+
+  props: {
+    datasetName: string,
+    isEditingMode: false,
+  }
+
+  state: {
+    dataset: ?APIDatasetType,
+    datasetJson: string,
+    isValidJSON: boolean,
+  } = {
+    dataset: null,
+    datasetJson: "",
+    isValidJSON: true,
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  async fetchData(): Promise<void> {
+    const datasetUrl = `/api/datasets/${this.props.datasetName}`;
+    const dataset = await Request.receiveJSON(datasetUrl);
+
+    const datasetJsonUrl = `${dataset.dataStore.url}/data/datasets/${this.props.datasetName}`;
+    const datasetJson = await Request.receiveJSON(datasetJsonUrl);
+
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState({
+      dataset,
+      datasetJson: JSON.stringify(datasetJson.dataSource, null, "  "),
+    });
+  }
+
+  importDataset = () => {
+    if (this.props.isEditingMode) {
+      const url = `/api/datasets/${this.props.datasetName}`;
+      Request.sendJSONReceiveJSON(url, {
```

Comment:
The first request just get's the `dataset` metadata from the wK main instance. In theory this should never fail (or otherwise the server is down). We also already use this request on the dashboard and the link to the "import view" shouldn't appear if something is truely wrong.
The second request tells the corresponding datastore to do the actual import operation.

The order is relevant. The 1. request contains information for building the url of the 2. request.